### DISCUSSION TITLE
v1.5

### DISCUSSION THREAD

**Discussion by lindesaysh** [lindesaysh/MRSea#35] [Announcements]
# Announcing the release of MRSea version 1.5

## Notes
* SALSA1D: 

  - function now allows the inclusion of a variable that is a `Date` class.
  
* SALSA2D:
  
  - the `getRadiiChoices` and `getRadiiChoices.vario` functions have been superseded and merged into one function: `getRadiiSequence()`.
  - functions have been added to help create a factor based knot grid (`selectFctrKnots`) and get an associated infinity-block distance matrix (updates to the `makeDists` function).  There is also a function to select space-filled starting knots for each factor level (`selectFctrStartk`)

* Vignettes:
  
  - Introduction to MRSea: added section on using `make.gamMRSea`
  - Interactions: edited to use the new functions for factor based knots.
  - Renewables Case Study: vignette re-instated and completely updated for all new functions. 
  - General updates to others on the basis on additional functionality
  
  
* Other: 

  - The seed used for CV score calculation is appended as an attribute to the output of cv.gamMRSea. Additionally it is also an optional input parameter. 
  - The seed used for generating the CV folds (`getCVids`) is now appended as an attribute for the returned vector of folds. 
  - the method for shortening the coefficient names in the summary output has been updated to be more efficient and to work in more situations. 
  
## Bug Fixes

* SALSA 1D:

* SALSA 2D:

  - The dispersion parameter estimate using `getDisperson` function was incorrectly specified as 1 for the Tweedie distribution.  This affected the dropping of "bad" knots in the initialise step resulting in more knots being dropped than should have been. This did not affect knots being subsequently added in the exchange step. The correct calculation is now included in the `getDispersion` function.
  - When using `initialise = FALSE` and specifying initial knot locations via `initialKnPos` code has been added to snap the initial locations to the nearest candidate knot locations in the knot grid. Previously it was assumed these initial locations were a subset of the candidate set. Now any locations may be used. 
  - When using an odd number of radii and with `chooseradii = FALSE` there may sometimes have been a mismatch from the output of SALSA2D and the output given from `runSALSA2D`. 
  

* Other:

  - When using the Tweedie distribution, the power parameter is not stored in the model call and so issues arise when using a stored model object where this parameter is no longer available.  The parameter stored within the model but updates have been made to `runSALSA1D` and `runSALSA2D` to ensure that the variance power and variance link are specified as numbers in the model call. 
  - The CV score was unchanged when changing K.  This occurred owing to the folds stored within the model object.  If `K` specified in `cv.gamMRSea` is different to that in the model object new folds are now created with a warning message.  The folds in the model object remain unchanged. Note that there is now an error message if the number of folds specified exceeds the number of unique panels. 
  - bug in the variance calculation for the gamma distribution in the `plotMeanVar` function. Additionally, the legend mismatched the data.
  - bug in the `runInfluence` and `timeInfluenceCheck`.  It worked fine for models without a 2D smooth but a specification error caused issues for models with one. Fixed and working as it should now. 
  

<hr /><em>This discussion was created from the release <a href='https://github.com/lindesaysh/MRSea/releases/tag/v1.5'>v1.5</a>.</em>

**Comment by dallasjordan** [comment]
Hi Lindesay, 

Thanks very much for your hard work on this very useful package. I wanted to clarify one of your bug fixes in your release notes for v1.5 - it looks like the SALSA 2D bug pertaining to the dispersion parameter estimate being 1 in getDispersion() was only present when used on a Tweedie model object (your commit changed phi = 1 for binomial and Tweedie models to just binomial models and added a separate phi calculation for Tweedie models). 

I believe this issue didn't affect quasipoisson models, but could you confirm that is the case, and perhaps update the bug fix notes to reflect this?

Many thanks,
Dallas

![image](https://github.com/lindesaysh/MRSea/assets/75545618/b15efa4d-8157-4338-ad44-c416a3c7a105)

**Reply by lindesaysh** [reply]
Hi Dallas,
Thanks for spotting this.  Yes it was only an issue for Tweedie models (which were only introduced at the end of August) and no other distributions. 
I have edited the text above and will change the news/changelog too but that will take a little longer to filter through to the website.
Thanks again,
Lindesay
